controllers/glanceapi_controller.go

```diff
@@ -1700,3 +1756,46 @@ func (r *GlanceAPIReconciler) GetHorizonEndpoint(
 
 	return ep, nil
 }
+
+// verifyApplicationCredentials checks if ApplicationCredential secret exists and adds it to configVars
+// The AC secret is created by the keystone-operator's AC controller when the AC is ready.
+// If the secret exists and is valid, we use AC auth. Otherwise, we fall back to password auth.
+func (r *GlanceAPIReconciler) verifyApplicationCredentials(
+	ctx context.Context,
+	_ *helper.Helper,
+	instance *glancev1.GlanceAPI,
+	configVars *map[string]env.Setter,
+) (ctrl.Result, error) {
+	log := r.GetLogger(ctx)
+
+	// Check if AC secret exists (created by keystone AC controller)
+	acSecretName := keystonev1.GetACSecretName(glance.ServiceName)
```

Comment:
`ServiceName` is a const that identifies the openstack image service [1] but this does not ensure this code will work if the CR name is different (e.g. `glance-12345678` instead of `glance`).
If we are looking for the current `glanceAPI` name here, we usually expose the `APIName()` function [2], so you can technically pass it to the function and consume it here (this would help to make the function generic and not have dependencies on `instance` type).  

[1] https://github.com/openstack-k8s-operators/glance-operator/blob/main/pkg/glance/const.go#L34
[2] https://github.com/openstack-k8s-operators/glance-operator/blob/main/api/v1beta1/glanceapi_types.go#L174